src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+        /// </remarks>
+        /// <param name="settings">The I2cConnectionSettings object initialized with the appropiate settings to communicate with the Ina219.</param>
+        public Ina219(I2cConnectionSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new System.ArgumentNullException(nameof(settings));
+            }
+
+            PowerLsb = CurrentLsb = 1F;
+            _calibrationValue = 0;
+
+            _disposeI2cDevice = true;
+            _i2cDevice = I2cDevice.Create(settings);
+        }
+
+        /// <summary>
+        /// Reset the Ina219 to default values;
+        /// </summary>
+        public void Reset()
+        {
+            ushort regValue;
+
+            // Reset the device by sending a value to the configuration register with the reset but set.
+            WriteRegister(Register.Configuration, (ushort)ConfigurationFlags.Rst);
+
+            // Read back the values from the configuration register which should now be default.
+            regValue = ReadRegister(Register.Configuration, 1000);
+
+            PowerLsb = PowerLsb = 1F;
+            _calibrationValue = 0;
+
+            // Get the current values for the Bus and Shunt ADC settings.
+            _busAdcResSamp = (AdcResolutionOrSamples)((regValue & (ushort)ConfigurationFlags.BadcMask) >> 4);
+            _shuntAdcResSamp = (AdcResolutionOrSamples)(regValue & (ushort)ConfigurationFlags.SadcMask);
+        }
+
+        /// <summary>
+        /// Set the Ina219 operating mode.
+        /// </summary>
+        /// <param name="mode">The operating mode to switch to.</param>
+        public void SetOperatingMode(OperatingMode mode)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.ModeMask ^ 0xFFFF;
+            regValue |= (ushort)mode;
+
+            WriteRegister(Register.Configuration, regValue);
+        }
+
+        /// <summary>
+        /// Set the Ina219 Bus voltage range.
+        /// </summary>
+        /// <param name="range">The voltage range to switch to.</param>
+        public void SetBusVoltageRange(BusVoltageRange range)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.BrngMask ^ 0xFFFF;
+            regValue |= (ushort)range;
+
+            WriteRegister(Register.Configuration, regValue);
+        }
+
+        /// <summary>
+        /// Set the Ina219 Programable Gain Amplifier range.
+        /// </summary>
+        /// <remarks>
+        /// This is the voltage range used to read the shunt voltage. It can be one of +/-40mV, +/-80mV, +/-160mV or +/-320mV.
+        /// </remarks>
+        /// <param name="sensitivity">The voltage range of the Programable Gain Amplifier to switch to.</param>
+        public void SetPGARange(PgaSensitivity sensitivity)
```

Comment:
ditto

Reply:
same for other stuff which just sets/reads register (method applies if we do any waiting or if value which you read changes between calls)

Reply:
I believe this is ‘waiting’ as it will take time to send the data down the I2c bus at a particular clock speed. And you may want an async version. If it was io mapped io or memory mapped io then I could see why a property would be right.

Reply:
@krwq 

I have pushed a new version of the INA219 binding and hopefully addressed your comments.